Deadlock in shellcheck integration on macOS: process.go:32 blocks writing to unstarted child stdin

### Summary
actionlint hangs (spinning ~450% CPU, never exits) on workflows that have **≥2 `run:` steps with ~1 KB+ shell scripts** when running on **darwin/arm64**. All shellcheck worker goroutines deadlock at `process.go:32` writing to the shellcheck child's stdin pipe.

### Reproduction

macOS arm64 (Darwin 25.4.0, M-series), shellcheck 0.11.0.

```yaml
# save as deadlock.yaml
name: N50
on: push
jobs:
  j:
    runs-on: ubuntu-latest
    steps:
      - run: |
          echo hello
          echo hello
          # … (50 identical `echo hello` lines)
      - run: |
          echo hello
          echo hello
          # … (50 identical `echo hello` lines)
```

```
$ actionlint deadlock.yaml
# hangs forever, ~450% CPU
```

Also reproduces on 1.7.10, 1.7.11, and 1.7.12 official darwin_arm64 release binaries (Go 1.25.5, 1.25.7, and 1.26.1 respectively) — so it's not a Go-version regression.

Running with `-shellcheck=` (integration disabled) finishes in ~20 ms. Running shellcheck standalone with the exact args actionlint passes (`--norc -f json -x --shell bash -e SC1091,SC2194,SC2050,SC2153,SC2154,SC2157,SC2043 -`) on the extracted scripts finishes in <100 ms, including 5 in parallel. So the hang is in actionlint, not shellcheck.

### Diagnosis

`SIGQUIT` stack trace (abridged) — every shellcheck worker goroutine is blocked here:

```
goroutine N [IO wait]:
internal/poll.(*pollDesc).waitWrite(...)
internal/poll.(*FD).Write
os.(*File).Write
io.WriteString
github.com/rhysd/actionlint.(*cmdExecution).run
    actionlint/process.go:32 +0x114
github.com/rhysd/actionlint.(*concurrentProcess).run.func1
    actionlint/process.go:99
golang.org/x/sync/errgroup.(*Group).Go.func1
```

The offending pattern in `process.go`:

```go
func (e *cmdExecution) run() ([]byte, error) {
    cmd := exec.Command(e.cmd, e.args...)
    cmd.Stderr = nil
    p, err := cmd.StdinPipe()            // pipe created, child NOT started
    ...
    if _, err := io.WriteString(p, e.stdin); err != nil { ... }   // ← blocks here
    p.Close()
    ...
    stdout, err = cmd.Output()           // cmd.Start() happens only here
}
```

Writing to `StdinPipe()` **before** calling `cmd.Start()` relies on the kernel pipe buffer to absorb the whole script. On darwin/arm64 the usable buffer against an unstarted child appears to be small enough (empirically <1 KB per pipe under contention) that any moderately sized script deadlocks. Once N≥2 workers all block pre-start, no reader ever attaches.

### Suggested fix

Let Go's stdlib handle stdin copying after `Start()`:

```go
func (e *cmdExecution) run() ([]byte, error) {
    cmd := exec.Command(e.cmd, e.args...)
    cmd.Stdin = strings.NewReader(e.stdin)    // stdlib spawns copier after Start
    if e.combineOutput {
        return cmd.CombinedOutput()
    }
    return cmd.Output()
}
```

`cmd.Output()` / `cmd.CombinedOutput()` internally start a goroutine to copy from `cmd.Stdin` to the child's stdin *after* `cmd.Start()`, which eliminates the pre-start write and is the canonical pattern in the Go stdlib docs.

### Environment

- actionlint 1.7.12 (also 1.7.11, 1.7.10) darwin_arm64
- shellcheck 0.11.0
- macOS Darwin 25.4.0, M-series Apple silicon
- Go 1.25.5 / 1.25.7 / 1.26.1 (per the release binary)

### Workaround

Pass `-shellcheck=` on the CLI, or run actionlint inside a Linux container (larger pipe buffers mask the bug).

--- a/process.go
+++ b/process.go
@@ -3,8 +3,8 @@ package actionlint
 import (
 	"context"
 	"fmt"
-	"io"
 	"os/exec"
+	"strings"
 	"sync"
 
 	"github.com/mattn/go-shellwords"
@@ -24,18 +24,14 @@ type cmdExecution struct {
 func (e *cmdExecution) run() ([]byte, error) {
 	cmd := exec.Command(e.cmd, e.args...)
 	cmd.Stderr = nil
-
-	p, err := cmd.StdinPipe()
-	if err != nil {
-		return nil, fmt.Errorf("could not make stdin pipe for %s process: %w", e.cmd, err)
-	}
-	if _, err := io.WriteString(p, e.stdin); err != nil {
-		p.Close()
-		return nil, fmt.Errorf("could not write to stdin of %s process: %w", e.cmd, err)
-	}
-	p.Close()
+	// Set stdin via an io.Reader so that exec.Cmd pipes the bytes to the child
+	// after Start(). Writing to cmd.StdinPipe() before Start() relies on the
+	// kernel pipe buffer being large enough to absorb the whole payload, which
+	// deadlocks on darwin when multiple workers run concurrently (issue #650).
+	cmd.Stdin = strings.NewReader(e.stdin)
 
 	var stdout []byte
+	var err error
 	if e.combineOutput {
 		stdout, err = cmd.CombinedOutput()
 	} else {

--- a/process_test.go
+++ b/process_test.go
@@ -176,6 +176,49 @@ func TestProcessInputStdin(t *testing.T) {
 	}
 }
 
+// Regression test for issue #650: concurrent runs with a stdin payload larger
+// than the kernel pipe buffer used to deadlock on darwin because the payload
+// was written to cmd.StdinPipe() before cmd.Start().
+func TestProcessConcurrentStdinDoesNotDeadlock(t *testing.T) {
+	p := newConcurrentProcess(5)
+
+	// 64 KiB is above the default pipe buffer size on darwin and Linux so it
+	// forces the stdin copy to happen after the child has started.
+	payload := strings.Repeat("x", 64*1024)
+
+	done := make(chan struct{})
+	go func() {
+		defer close(done)
+		cmds := make([]*externalCommand, 0, 5)
+		for i := 0; i < 5; i++ {
+			cat := testSkipIfNoCommand(t, p, "cat")
+			cat.run(nil, payload, func(b []byte, err error) error {
+				if err != nil {
+					t.Errorf("cat failed: %v", err)
+					return err
+				}
+				if len(b) != len(payload) {
+					t.Errorf("cat output length %d, want %d", len(b), len(payload))
+				}
+				return nil
+			})
+			cmds = append(cmds, cat)
+		}
+		for _, c := range cmds {
+			if err := c.wait(); err != nil {
+				t.Errorf("cat wait failed: %v", err)
+			}
+		}
+		p.wait()
+	}()
+
+	select {
+	case <-done:
+	case <-time.After(10 * time.Second):
+		t.Fatal("concurrent stdin writes deadlocked — see issue #650")
+	}
+}
+
 func TestProcessErrorCommandNotFound(t *testing.T) {
 	p := newConcurrentProcess(1)
 	c := &externalCommand{

